Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Scripts/Interactable.cs

```diff
@@ -221,149 +262,258 @@ public List<InteractableEvent> InteractableEvents
+        {
+            get { return GetStateValue(InteractableStates.InteractableStateEnum.VoiceCommand) > 0; }
+            set { SetState(InteractableStates.InteractableStateEnum.VoiceCommand, value); }
+        }
 
         /// <summary>
         /// A near interaction touchable is actively being touched
         /// </summary>
-        public bool HasPhysicalTouch { get; private set; }
+        public virtual bool HasPhysicalTouch
+        {
+            get { return GetStateValue(InteractableStates.InteractableStateEnum.PhysicalTouch) > 0; }
+            set { SetState(InteractableStates.InteractableStateEnum.PhysicalTouch, value); }
+        }
 
         /// <summary>
-        /// Misc - custom: not set by Interactable
+        /// State that corresponds to miscellaneous/custom use by consumers
+        /// Currently not controlled by Interactable directly
         /// </summary>
-        public bool HasCustom { get; private set; }
+        public virtual bool HasCustom
+        {
+            get { return GetStateValue(InteractableStates.InteractableStateEnum.Custom) > 0; }
+            set { SetState(InteractableStates.InteractableStateEnum.Custom, value); }
+        }
 
         /// <summary>
-        /// A near interaction grabbable is actively being grabbed/
+        /// A near interaction grabbable is actively being grabbed
         /// </summary>
-        public bool HasGrab { get; private set; }
-
-        // internal cached states
-        protected State lastState;
-        protected bool wasDisabled = false;
+        public virtual bool HasGrab
+        {
+            get { return GetStateValue(InteractableStates.InteractableStateEnum.Grab) > 0; }
+            set { SetState(InteractableStates.InteractableStateEnum.Grab, value); }
+        }
 
         #endregion
 
-        // check for isGlobal or RequiresFocus changes
-        protected bool requiresFocusValueCheck;
-        protected bool isGlobalValueCheck;
+        protected State lastState;
 
-        // cache of current dimension
-        [SerializeField]
-        protected int dimensionIndex = 0;
+        // directly manipulate a theme value, skip blending
+        protected bool forceUpdate = false;
 
         // allows for switching colliders without firing a lose focus immediately
         // for advanced controls like drop-downs
         protected float rollOffTime = 0.25f;
         protected float rollOffTimer = 0.25f;
 
-        // cache voice commands
         protected string[] voiceCommands;
 
-        // IInteractableEvents
         protected List<IInteractableHandler> handlers = new List<IInteractableHandler>();
 
         protected Coroutine globalTimer;
 
-        #region Click Variables
-
         // A click must occur within this many seconds after an input down
         protected float clickTime = 1.5f;
         protected Coroutine clickValidTimer;
-        protected int clickCount = 0;
         protected float globalFeedbackClickTime = 0.3f;
```

Comment:
>protected float globalFeedbackClickTime = 0.3f; [](start = 8, length = 47)

nit: it should either be editor configurable or a const

Reply:
I don't want to show it in editor. It'll just confuse people and most don't need access to it. 

I didn't want to make it const because it's protected. Someone may inherit this class and set a new value for the property

Reply:
Could you add a comment to what the property is used for? (i have to go skim through code to understand how clickTime and globalFeedbackClickTime are different)

---
In reply to: [329207521](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/6104#discussion_r329207521) [](ancestors = 329207521)